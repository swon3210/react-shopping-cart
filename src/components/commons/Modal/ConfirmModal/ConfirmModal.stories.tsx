import ConfirmModal, { Props } from './ConfirmModal';

export default {
  title: 'Components/Commons/Modal/ConfirmModal',
  component: ConfirmModal,
};

const Template = (args: Props) => <ConfirmModal {...args} />;

export const Default = Template.bind({});
(Default as any).args = {};
