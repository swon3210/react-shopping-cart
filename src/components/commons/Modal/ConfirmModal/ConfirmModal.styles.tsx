import styled from 'styled-components';

export const ModalContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 129px;
`;

export const Text = styled.span`
  font-size: 35px;
  line-height: 51px;
  text-align: center;
  letter-spacing: 0.7px;
  margin-bottom: 92px;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;

  & > button:not(:last-child) {
    margin-right: 24px;
  }
`;
