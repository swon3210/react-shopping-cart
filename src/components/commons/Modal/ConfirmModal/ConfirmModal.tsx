import * as Styled from './ConfirmModal.styles';
import Modal from '../Modal';
import Button from '../../Button/Button';
import { COLORS } from '../../../../constants';

export interface Props {
  heading: string;
  onClose?: () => void;
  cancelButtonText: string;
  onCancel?: () => void;
  confirmButtonText: string;
  onConfirm?: () => void;
}

const ConfirmModal = ({
  heading = '텍스트를 입력해주세요',
  onClose,
  onCancel,
  cancelButtonText = '취소',
  onConfirm,
  confirmButtonText = '확인',
}: Props) => {
  return (
    <Modal onBackDropClick={onClose}>
      <Styled.ModalContentWrapper>
        <Styled.Text>{heading}</Styled.Text>
        <Styled.ButtonWrapper>
          <Button size="MD" onClick={onCancel} backgroundColor={COLORS.GRAY_300}>
            {cancelButtonText}
          </Button>
          <Button size="MD" onClick={onConfirm}>
            {confirmButtonText}
          </Button>
        </Styled.ButtonWrapper>
      </Styled.ModalContentWrapper>
    </Modal>
  );
};

export default ConfirmModal;
