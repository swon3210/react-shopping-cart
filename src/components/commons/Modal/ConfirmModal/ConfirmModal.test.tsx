import ConfirmModal from './ConfirmModal';
import { render } from '@testing-library/react';

describe('ConfirmModal Component', () => {
  it('ConfirmModal Snapshot', () => {
    const confirmModalUtil = render(
      <ConfirmModal heading="테스트 헤딩" cancelButtonText="취소" confirmButtonText="확인" />
    );
    expect(confirmModalUtil.asFragment()).toMatchSnapshot();
  });
});
