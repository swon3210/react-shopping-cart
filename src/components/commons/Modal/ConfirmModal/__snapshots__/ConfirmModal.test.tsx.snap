// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal Component ConfirmModal Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-hKwCoD hpQaIv"
  >
    <div
      class="sc-jRQAMF ifTFhu"
    />
    <div
      class="sc-eCImvq gbwWIx"
    >
      <div
        class="sc-bdvvaa bGeHZK"
      >
        <span
          class="sc-gsDJrp fRcINU"
        >
          테스트 헤딩
        </span>
        <div
          class="sc-dkPtyc gdurRt"
        >
          <button
            class="sc-gKckTs fAvngU"
            type="button"
          >
            취소
          </button>
          <button
            class="sc-gKckTs jMTFfc"
            type="button"
          >
            확인
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
